### ISSUE TITLE
Bug: Particle.static_residuals returns all-NaN when particle epoch equals first observation time

### ISSUE THREAD

**Issue by ben-cassese** [ben-cassese/jorbit#64]
# Bug: `Particle.static_residuals` returns all-NaN when particle epoch equals first observation time

## Symptoms

Calling `p.static_residuals(p.cartesian_state)` (or any CartesianState) returns an array of `nan` when the particle is initialised at the time of its first observation — the most common usage.

## Minimal reproduction

```python
import jax
jax.config.update("jax_enable_x64", True)
import jax.numpy as jnp
import astropy.units as u
from astropy.time import Time
from astropy.coordinates import SkyCoord
from astroquery.jplhorizons import Horizons
from jorbit import Particle, Observations

nights = [Time("2025-01-01 07:00"), Time("2025-01-02 07:00"), Time("2025-01-05 07:00")]
times = []
for n in nights:
    times.extend([n + i * 1 * u.hour for i in range(3)])
times = Time(times)

obj = Horizons(id="274301", location="695@399", epochs=times.utc.jd)
pts = obj.ephemerides(extra_precision=True, quantities="1")
coords = SkyCoord(pts["RA"], pts["DEC"], unit=(u.deg, u.deg))
times = Time(pts["datetime_jd"], format="jd", scale="utc")

obs = Observations(
    observed_coordinates=coords,
    times=times,
    observatories="kitt peak",
    astrometric_uncertainties=1 * u.arcsec,
)

obj = Horizons(id="274301", location="500@0", epochs=times.tdb.jd[0])
vecs = obj.vectors(refplane="earth")
true_x0 = jnp.array([vecs["x"], vecs["y"], vecs["z"]]).T[0]
true_v0 = jnp.array([vecs["vx"], vecs["vy"], vecs["vz"]]).T[0]

p = Particle(x=true_x0, v=true_v0, time=times[0], observations=obs)
print(p.static_residuals(p.cartesian_state))
# → [[nan nan] [nan nan] [nan nan] [nan nan] [nan nan] [nan nan] [nan nan] [nan nan] [nan nan]]
```

## Root cause

`precompute_likelihood_data` in `src/jorbit/likelihoods/setup_static_likelihood.py` seeds the IAS15 adaptive integrator with:

```python
integrator_init.dt = obs_times[0] - t0 if len(obs_times) > 0 else 10.0
```

Observation times are stored as offsets from the particle's reference epoch (`t_ref`). When the particle is initialised at the first observation time (the typical pattern), `t_ref = times[0]`, so `obs_times[0] = 0.0` and `t0 = 0.0`. The seed step size is therefore `0.0`.

IAS15 divides by `dt` when computing its local truncation error estimate. With `dt = 0`, this immediately produces `nan`, which propagates into every element of the precomputed `dts` array. Because `dts` is frozen into the `static_residuals` JIT closure at Particle construction time, every subsequent call to `static_residuals` returns `nan` regardless of the input state.

The bug was introduced during the time-rebasing refactoring that moved JAX-visible times from absolute JD to offsets from `t_ref`. The seed-dt logic was not updated to handle the case where the particle epoch and the first observation are at the same time.

**Not affected:** `Particle.loglike`, `Particle.scipy_objective`, and `Particle.max_likelihood` all use the dynamic integrator path, which does not go through `precompute_likelihood_data`.

## Fix

In `precompute_likelihood_data` (`src/jorbit/likelihoods/setup_static_likelihood.py`), fall back to the first inter-observation spacing when the epoch-to-first-obs gap is ≤ 0:

```python
# Before (broken when obs_times[0] == t0 == 0):
integrator_init.dt = obs_times[0] - t0 if len(obs_times) > 0 else 10.0

# After:
dt_seed = float(obs_times[0] - t0) if len(obs_times) > 0 else 10.0
if dt_seed <= 0.0:
    dt_seed = float(obs_times[1] - obs_times[0]) if len(obs_times) > 1 else 1.0
integrator_init.dt = dt_seed
```

The seed value is only a hint to the adaptive stepper and does not affect output accuracy.

A regression test (`test_static_residuals` in `tests/test_particle.py`) has been added to catch this class of issue going forward.
